### PR TITLE
Implement restart policy for VMs

### PR DESCRIPTION
**Pull request by edigaryev**
Resolves https://github.com/cirruslabs/orchard/issues/68.

### CODE REVIEW

File: internal/worker/worker.go [hunk 1]
```diff
@@ -29,13 +29,15 @@ type Worker struct {
 	syncRequested chan bool
 	vmm           *vmmanager.VMManager
 	client        *client.Client
+	pollTicker    *time.Ticker
```
Comment (fkorotkov) [inline internal/worker/worker.go:32]: Why did you add it?

Comment (edigaryev) [reply internal/worker/worker.go:32]: We had no time-based rate limiter in the `Worker.Run()` loop, so I've added it there, and re-used the ticker this way.

File: internal/controller/scheduler/scheduler.go [hunk 2]
```diff
@@ -205,6 +209,25 @@ func (scheduler *Scheduler) healthCheckingLoopIteration() error {
...
 func (scheduler *Scheduler) healthCheckVM(txn storepkg.Transaction, nameToWorker map[string]v1.Worker, vm v1.VM) error {
+	logger := scheduler.logger.With("vm_name", vm.Name, "vm_uid", vm.UID, "vm_restart_count", vm.RestartCount)
+
+	// Schedule a VM restart if the restart policy mandates it
+	needsRestart := vm.RestartPolicy == v1.RestartPolicyOnFailure &&
+		vm.Status == v1.VMStatusFailed &&
+		time.Since(vm.RestartedAt) > schedulerVMRestartDelay
+
+	if needsRestart {
+		logger.Debugf("restarting VM")
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply internal/controller/scheduler/scheduler.go:220]: It's already added to a logger just a couple lines above (at the beginning of the function).

**Review by fkorotkov — APPROVED** [review]
